components/webln.js

```diff
@@ -0,0 +1,106 @@
+import { createContext, createRef, useContext, useEffect, useImperativeHandle, useState } from 'react'
+
+const WebLNContext = createContext({})
+export const WebLNContextRef = createRef()
+
+const fetchWebLNProvider = async () => {
+  // sync provider from local storage
+  // TODO check if there is a better, more secure option
+  const configJson = window.localStorage.getItem('bc:config')
+  if (configJson) {
+    const WebLNProviders = await import('@getalby/bitcoin-connect').then((mod) => mod.WebLNProviders)
+    const config = JSON.parse(configJson)
+    if (config.connectorType === 'lnbits') return new WebLNProviders.LnbitsWebLNProvider(config.lnbitsInstanceUrl, config.lnbitsAdminKey)
+    // TODO implement other connector types
+  }
+}
+
+export function WebLNProvider ({ children }) {
+  const [provider, setProvider] = useState(null)
+  const [info, setInfo] = useState(null)
+  const [balance, setBalance] = useState(0)
+
+  const initProvider = async (provider) => {
+    const WebLNProviders = await import('@getalby/bitcoin-connect').then((mod) => mod.WebLNProviders)
+    if (provider instanceof WebLNProviders.LnbitsWebLNProvider) {
+      const connector = 'lnbits'
+      // TODO
+      //   we're reaching into private parts here.
+      //   is there a better way to show to user which connection was used?
```

Comment:
What is exactly needed here? ideally `webln.getInfo()` returns the necessary info. the website should only interact with the WebLN protocol as much as possible

Reply:
I am only doing this because I couldn't get `onConnected` to fire on page load. I would love to not do this, haha

That's what I meant with "why do I have to do this myself?" in the TODO in `onConnected`:

> // TODO
//   why is this not fired on page load when there is a provider available in local storage?
//   why do i have to do this myself?

Reply:
`webln.getInfo()` (I use `provider.getInfo()` but I think this makes no difference and I think that's what you meant, too) returns this:

```
{
    "node": {
        "alias": "ek",
        "pubkey": ""
    },
    "methods": [
        "getInfo",
        "getBalance",
        "sendPayment"
    ],
    "version": "1.0",
    "supports": [
        "lightning"
    ]
}
```

I thought it would be nice to show the user which wallet they connected.

![2023-12-28-135030_333x245_scrot](https://github.com/stackernews/stacker.news/assets/27162016/c61db60d-5cff-4ad1-acc1-58a8f2fb6b15)

Reply:
If you click the connected button it will show where you are connected (via lnbits connector), and actually the LNbits hostname is stored in the bitcoin connect config. But does it help if you know the exact hostname, or would it be enough to know you're connected via LNbits?

I created an issue here: https://github.com/getAlby/bitcoin-connect/issues/177 - maybe we can figure a way to expose some helpful info in a generic way for all connectors.

Reply:
> But does it help if you know the exact hostname, or would it be enough to know you're connected via LNbits?

Mhh, I think it might be enough to know that I am connected to LNbits. Not entirely sure yet though. When we released the first iteration on this, user feedback will be more valuable than any feedback I can provide you on this as a developer.

Reply:
I removed the code that is reaching into private parts since https://github.com/getAlby/bitcoin-connect/issues/177 should fix the use case I was solving for.